pkg/machines/vmnetworktab.jsx

```diff
@@ -0,0 +1,162 @@
+/*jshint esversion: 6 */
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React from 'react';
+import cockpit from 'cockpit';
+import { changeNetworkState } from "./actions.es6";
+import { rephraseUI, vmId } from "./helpers.es6";
+import { Listing, ListingRow } from 'cockpit-components-listing.jsx';
+
+const _ = cockpit.gettext;
+
+class VmNetworkTab extends React.Component {
+    render () {
```

Comment:
I don't think we have any hard fast rules on this but, is there any way we can split this up a little so that there is a little less mixing of html and js. I think it might make it a little easier to follow. 

Also would be nice if indention of opening and closing () matched everwhere. 

@larskarlitski or @martinpitt any thoughts?

Reply:
We don't have a rule, but I agree that this is a bit too intermingled. I've found that splitting the logic from the rendering makes the code much easier to understand for other people.